fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -131,7 +133,14 @@ public LuceneIndexMaintainer(@Nonnull final IndexMaintainerState state, @Nonnull
                 parser = new QueryParser(PRIMARY_KEY_SEARCH_NAME, analyzer);
             }
             Query query = parser.parse(range.getLow().getString(0));
-            return new LuceneRecordCursor(executor, scanProperties, state, query, continuation, state.index.getRootExpression().normalizeKeyForPositions());
+            String groupingKey = null;
+            for (int i = 1; i < range.getLow().getItems().size(); i++) {
+                Object comparison = range.getLow().get(i);
```

Comment:
I believe that some of the possible value types do not have equality-preserving `toString`. The one that occurs to me right away is byte array, which will be `[B...`.